src/components/profile/RepositoryItem.jsx

```diff
@@ -7,50 +9,196 @@ import CardContent from '@material-ui/core/CardContent';
+import Edit from '@material-ui/icons/Edit';
+import Delete from '@material-ui/icons/Delete';
 
-const RepositoryItem = ({
-  name,
-  description,
-  owner,
-  htmlUrl,
-  sshUrl,
-  gitUrl,
-  pypiName,
-}) => {
-  return(
-  <ListItem>
-    <Card style={{width: '100%'}}>
-      <div style={{display: 'flex', flexFlow: 'row wrap'}}>
-        <div style={{flex: '1 1 10px', margin: 'auto', top: '0', bottom: '0', textAlign: 'center'}}>
-          {/* add pictures */}
-          <SciGymLogo /> 
-        </div>
-        <div style={{flex: '1 1 400px'}}>
-        <CardContent>
-          <Typography variant="h5" component='h2' gutterBottom>
-            {name}
-          </Typography>
-          <Typography variant='subheading' gutterBottom>
-            {description}
-          </Typography>
-          <Typography variant='subheading' gutterBottom>
-            Owner: <a href={"https://github.com/".concat(owner.username)}> {owner.username} </a>
-          </Typography>
-          <Typography variant="subheading">{pypiName && <pre>pip install {pypiName}</pre>}</Typography>
-        </CardContent>
-        <CardActions>
-          <Button href={htmlUrl}>
-            <GithubIcon />
-            Github
-          </Button>
-        </CardActions>
-        </div>
-      </div>
-
-    </Card>
-  </ListItem>
-  )};
-
-export default RepositoryItem;
+import { withStyles } from "@material-ui/core";
+import { compose } from 'redux';
+import EnvironmentForm from "./EnvironmentForm";
+import DeleteEnvironment from "./DeleteEnvironment";
+import api from '../../utils/api';
+import { getEnvironments } from "../../actions/environments";
+
+
+const styles = theme => ({
+  root: {
+    display: 'flex', 
+    flexFlow: 'row wrap',
+  },
+  cardStyle: {
+    width: '100%',
+  },
+  logoStyle: {
+    flex: '1 1 10px', 
+    margin: 'auto', 
+    top: '0', 
+    bottom: '0',
+    textAlign: 'center',
+  },
+  cardContentStyle: {
+    flex: '1 1 400px',
+  },
+  buttonStyle: {
+    margin: theme.spacing.unit,
+  },
+  buttonPosition: {
+    position: 'absolute',
+    right: '40px',
+    bottom: '25px',
+  }
+})
+
+class RepositoryItem extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      envExists: Boolean(this.props.environment),
+      openDelete: false,
+      error: '',
+    };
+  }
+
+  handleClickOpen = () => {
+    this.setState({
+      open: true,
+    });
+  }
+
+  handleClose = () => {
+    this.setState({ open: false });
+  }
+
+  handleClickDelete = () => {
+    this.setState({openDelete: true})
+  }
+
+  handleCloseDelete = () => {
+    this.setState({openDelete: false})
+  }
+
+  handleDelete = () => {
+    api.deleteEnvironment(this.props.environment).then(this.handleSuccess).catch(this.handleFailure);
+  }
+
+  handleSuccess = () => {
+    api.environments().then((json) => {
+      this.props.getEnvironments(json.data);
+      this.setState( {envExists: ! this.state.envExists, openDelete: false} );
```

Comment:
nitpicky, but some inconsistent spacing with parenthesis/brackets

Reply:
pleae be nitpicky :)